libraries/timeslider/src/components/TimeSlider/designer.ts

```diff
@@ -0,0 +1,223 @@
+import {
+    applyComponentModelDesignerSettings,
+    ApplyDesignerSettingsCallback,
+    ComponentModelDesignerSettings,
+    DesignerSettings,
+    getComponentModelDesignerSettings,
+    getComponentModelDesignerSettingsSchema,
+    GetDesignerSettingsCallback,
+    GetDesignerSettingsSchemaCallback,
+    Setting,
+    SettingsSchema,
+} from "@vertigis/web/designer";
+
+import TimeSliderModel from "./TimeSliderModel";
+
+export type TimeSliderLayout = "auto" | "compact" | "wide";
+export type TimeSliderMode =
+    | "instant"
+    | "time-window"
+    | "cumulative-from-start"
+    | "cumulative-from-end";
+export type TimeIntervalUnit =
+    | "milliseconds"
+    | "seconds"
+    | "minutes"
+    | "hours"
+    | "days"
+    | "weeks"
+    | "months"
+    | "years"
+    | "decades"
+    | "centuries";
+
+export interface TimeSliderSettings extends ComponentModelDesignerSettings {
+    layout: TimeSliderLayout;
+    loop: boolean;
+    playRate: number;
+    mode: TimeSliderMode;
+    overrideStops: boolean;
+    timeInterval: number;
+    timeIntervalUnit: TimeIntervalUnit;
+    timeVisible: boolean;
+}
+
+export type SettingsMap = DesignerSettings<TimeSliderSettings>;
+
+export const applySettings: ApplyDesignerSettingsCallback<
+    TimeSliderModel,
+    SettingsMap
+> = async (args) => {
+    const { model, settings } = args;
+    await applyComponentModelDesignerSettings(args);
+    model.assignProperties(settings);
+};
+
+export const getSettings: GetDesignerSettingsCallback<
+    TimeSliderModel,
+    SettingsMap
+> = async (args) => {
+    const { model } = args;
+    const {
+        layout,
+        loop,
+        playRate,
+        mode,
+        overrideStops,
+        timeInterval,
+        timeIntervalUnit,
+        timeVisible,
+    } = model;
+    return {
+        ...(await getComponentModelDesignerSettings(args)),
+        layout,
+        loop,
+        playRate,
+        mode,
+        overrideStops,
+        timeInterval,
+        timeIntervalUnit,
+        timeVisible,
+    };
+};
+
+export const getSettingsSchema: GetDesignerSettingsSchemaCallback<
+    TimeSliderModel,
+    SettingsMap
+> = async (args) => {
+    const baseSchema = await getComponentModelDesignerSettingsSchema(args);
+    (baseSchema.settings[0].settings as Setting<TimeSliderSettings>[]) = (
+        baseSchema.settings[0].settings as Setting<TimeSliderSettings>[]
+    ).concat([
+        {
+            id: "layout",
+            type: "select",
+            values: [
+                {
+                    displayName: "Compact",
```

Comment:
The display names for the options should also be turned into language keys so that they can be translated.